api/receiver/azure.go

```diff
@@ -35,65 +46,80 @@ func newAzureArtifact(artifactName string, zip io.Reader) (azureArtifact, error)
 	}, err
 }
 
-// extractReports extracts report files from the given zip.
-func (a azureArtifact) getReportFiles() ([]*zip.File, error) {
+// getFilesMatchingPattern returns the number and a slice of *zip.File whose
+// filenames match the given pattern.
+func (a azureArtifact) getFilesMatchingPattern(p *regexp.Regexp) (int, []*zip.File, error) {
 	if a.z == nil {
 		r, err := zip.NewReader(bytes.NewReader(a.data), int64(len(a.data)))
 		if err != nil {
-			return nil, err
+			return 0, nil, err
 		}
 		a.z = r
 	}
 	files := make([]*zip.File, 0, len(a.z.File))
 	for _, f := range a.z.File {
-		if reportPathRegex.MatchString(f.Name) {
+		if p.MatchString(f.Name) {
 			files = append(files, f)
 		}
 	}
-	return files, nil
+	return len(files), files, nil
```

Comment:
Why is the caller not able to len the array themselves?

Reply:
For convenience, so that its only caller can directly assign the return value to `results, resultFiles, err` (or screenshots) in one line. I wouldn't have written it this way if the function was more generic.